### PR TITLE
Add student test

### PR DESCRIPTION
**Pull request by Bernardstanislas**

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+  - mocha
+  - chai-friendly
 rules:
-  'no-console': error
+  "no-console": error
+  "mocha/no-exclusive-tests": error
+  "mocha/no-skipped-tests": error
+  "mocha/no-identical-title": error
+  "no-unused-expressions": off
+  "chai-friendly/no-unused-expressions": error
```
Comment (rdubigny) [inline .eslintrc.yml]: while we are at linting code can we requires trailing commas ? https://eslint.org/docs/rules/comma-dangle

File: api/student/student.api.test.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+const serverTest = require('../test/utils/server')
+const nock = require('nock')
+
+describe('Etudiant API', function () {
+  const server = serverTest()
+  const api = server.api
+  const validIne = 'fakeIne'
+  const invalidIne = 'invalidIne'
+  const apiKey = '3841b13fa8032ed3c31d160d3437a76a'
```
Comment (rdubigny) [inline api/student/student.api.test.js]: no need to put actual secret in tests.

File: api/student/student.api.test.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  nock('http://194.57.7.166/supdata/api/rest.php')
```
Comment (rdubigny) [inline api/student/student.api.test.js]: same remarks for this IP adress

File: api/student/student.controller.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    this.baseUrl = options.supdataHost
+    this.apiKey = options.supdataApiKey
+    this.authorize = this.authorize.bind(this)
+    this.ping = this.ping.bind(this)
+    this.student = this.student.bind(this)
+    this.consumerMatch = this.consumerMatch.bind(this)
+  }
+
+  authorize (req, res, next) {
+    if (req.authType === 'FranceConnect') {
```
Comment (rdubigny) [inline api/student/student.controller.js]: "c'est fou !"

Comment (rdubigny) [reply api/student/student.controller.js]: You shouldn't design something for FranceConnect at this point.

File: .eslintrc.yml [hunk 5]
```diff
@@ -1,5 +1,15 @@
...
+plugins:
+  - mocha
+  - chai-friendly
 rules:
-  'no-console': error
+  "no-console": error
+  "mocha/no-exclusive-tests": error
+  "mocha/no-skipped-tests": error
+  "mocha/no-identical-title": error
+  "no-unused-expressions": off
```
[the remark this answers is not in the document]
Comment (Bernardstanislas) [reply .eslintrc.yml]: as seen together, the linting rules have not changed, I just grouped theme

File: .eslintrc.yml (hunk 1 above)
Comment (Bernardstanislas) [reply .eslintrc.yml]: As discussed, I'd like not to change the linting rules until I'm fully familiar with the project

File: api/student/student.controller.js (hunk 4 above)
Comment (Bernardstanislas) [reply api/student/student.controller.js]: 🙆‍♂

File: api/student/student.api.test.js [hunk 6]
```diff
@@ -6,13 +6,13 @@ describe('Etudiant API', function () {
   const api = server.api
   const validIne = 'fakeIne'
   const invalidIne = 'invalidIne'
-  const apiKey = '3841b13fa8032ed3c31d160d3437a76a'
+  const apiKey = 'georges-moustaki'
```
Comment (rdubigny) [inline api/student/student.api.test.js]: gender offense

**Comment by rdubigny** [comment]
seen IRL, the token authorization is still missing in this PR

**Pull request merged**